synapse_invite_checker/responses.py

```diff
@@ -0,0 +1,91 @@
+# Copyright (C) 2025 Famedly
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+from dataclasses import dataclass
+
+from synapse_invite_checker.config import InsuredOnlyRoomScanConfig
+
+
+# This belongs in types.py, but due to circular imports it does not like living there.
+# Between DefaultPermissionConfig and InsuredOnlyRoomScanConfig, they are not happy.
+# This place is better than sticking it in config.py to make the imports happy
+@dataclass(slots=True)
+class EpaRoomTimestampResults:
+    """
+    Collection of timestamps and helpers to decide if a room should have members removed.
+    Used for EPA server rooms. Recall that invite/leave timestamps are *only* seeded
+    from pro users membership events
+
+    Attributes:
+        config: The insured only room scan config section from the invite checker
+        last_invite_in_room: The timestamp for the newest 'invite' membership in the
+            room. Only used if there was no detected 'leave' event.
+        last_leave_in_room: The timestamp of the newest 'leave' membership in the room.
+            The preferred timestamp to acquire.
+        room_creation_ts: If nothing else, use the room creation as a sentinel. The slim
+            possibility exists that no 'leave' event exists and any 'invite' may have
+            failed. This is allowed to give a user time to try again before they do not
+            have access to this room removed.
+    """
+
+    config: InsuredOnlyRoomScanConfig
```

Comment:
if you make this just a basic class and don't put the config in here, you can put it in the types.py. I don't think you need to make this class this fancy, since it is only used in one place.